src/libraries/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedCmsTests.netcoreapp.cs

```diff
@@ -844,7 +1100,18 @@ public static void ExistingDocument_Ecdsa_Sha256_FromNetFX()
             Assert.Equal(Oids.EcPublicKey, signerInfo.SignatureAlgorithm.Value);
         }
 
-        private static void VerifyWithExplicitPrivateKey(X509Certificate2 cert, AsymmetricAlgorithm key)
+        private delegate CmsSigner CreateSignerFunc<TKey>(SubjectIdentifierType sit, X509Certificate2 cert, TKey key);
+        private static CreateSignerFunc<AsymmetricAlgorithm> CreateAsymmetricAlgorithmSigner = (sit, cert, key) =>
+        {
+            return new CmsSigner(sit, cert, key);
+        };
+
+        private static CreateSignerFunc<SlhDsa> CreateSlhDsaSigner = (sit, cert, key) =>
+        {
+            return new CmsSigner(sit, cert, key);
+        };
```

Comment:
I don't think these delegates add value (but do add cost and complexity).

VerifyWithExplicitPrivateKey should just take `object` and type test to know what to call.

And then I don't have to be upset about the chopping style used where these made the lines too long :smile: